pmd-core/src/main/java/net/sourceforge/pmd/properties/builders/SingleNumericPropertyBuilder.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.properties.builders;
+
+/**
+ * @author Clément Fournier
+ * @since 6.0.0
+ */
+public abstract class SingleNumericPropertyBuilder<V, T extends SingleNumericPropertyBuilder<V, T>>
+        extends SingleValuePropertyBuilder<V, T> {
+
+
+    protected V lowerLimit;
+    protected V upperLimit;
+
+
+    /**
+     * Specify a minimum value.
+     *
+     * @param val Value
+     *
+     * @return The same builder
+     */
+    @SuppressWarnings("unchecked")
+    public T min(V val) {
```

Comment:
considering `NumericPropertyModule` requires both, `lowerLimit` and `upperLimit` to be set, we may want to add a `T range(V min, V max)` method to reduce boilerplate a little further.